From fef736ef9bb7af5266e89247450beb1439f2187c Mon Sep 17 00:00:00 2001
From: Vu Dang <vu.dang.te@renesas.com>
Date: Wed, 11 Sep 2024 03:37:11 +0000
Subject: [PATCH] meta: classes: esdk: explicitly address the location of
 devtool by alias

This patch is a workaround for the environment script to point to
the location of devtool in sysroots by using alias as sysroots's hardcoded
install location does not change when setting up eSDK.

Signed-off-by: Vu Dang <vu.dang.te@renesas.com>
---
 meta/classes/populate_sdk_ext.bbclass | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/meta/classes/populate_sdk_ext.bbclass b/meta/classes/populate_sdk_ext.bbclass
index 1bdfd92847..15a410f091 100644
--- a/meta/classes/populate_sdk_ext.bbclass
+++ b/meta/classes/populate_sdk_ext.bbclass
@@ -670,7 +670,13 @@ sdk_ext_postinst() {
 
 	# A bit of another hack, but we need this in the path only for devtool
 	# so put it at the end of $PATH.
-	echo "export PATH=\"$target_sdk_dir/sysroots/${SDK_SYS}${bindir_nativesdk}:\$PATH\"" >> $env_setup_script
+	#echo "export PATH=\"$target_sdk_dir/sysroots/${SDK_SYS}${bindir_nativesdk}:\$PATH\"" >> $env_setup_script
+	
+	# Workaround: This sysroots does not work due to its hardcoded install location does not change when setting up eSDK.
+	# So, we only use the sysroots from the Yocto build tools (buildtools).
+	# And as this path is only for devtool, address directly to devtool location in this sysroots by using alias.
+	# It's a hack to make sure that the devtool is always there when preparing environment for eSDK.
+	echo "alias devtool=\"$target_sdk_dir/sysroots/${SDK_SYS}${bindir_nativesdk}/devtool\"" >> $env_setup_script
 
 	echo "printf 'SDK environment now set up; additionally you may now run devtool to perform development tasks.\nRun devtool --help for further details.\n'" >> $env_setup_script
 
-- 
2.25.1

